src/Farmer/Builders/Builders.HostGroup.fs

```diff
@@ -0,0 +1,185 @@
+﻿[<AutoOpen>]
+module Farmer.Builders.HostGroups
+
+open System
+open Farmer
+open Farmer.Arm
+open Farmer.DedicatedHosts
+
+type HostGroupConfig =
+    {
+        Name: ResourceName
+        AvailabilityZones: AvailabilityZone list option
+        SupportAutomaticPlacement: FeatureFlag option
+        PlatformFaultDomainCount: PlatformFaultDomainCount option
+        Tags: Map<string, string>
+    }
+
+    interface IBuilder with
+        member this.ResourceId = hostGroups.resourceId this.Name
+
+        member this.BuildResources location =
+            let hostGroup: HostGroup =
+                {
+                    Name = this.Name
+                    Location = location
+                    AvailabilityZones = this.AvailabilityZones |> (Option.defaultValue List.Empty)
+                    SupportAutomaticPlacement =
+                        this.SupportAutomaticPlacement |> (Option.defaultValue FeatureFlag.Disabled)
+                    PlatformFaultDomainCount =
+                        this.PlatformFaultDomainCount
+                        |> (Option.defaultValue PlatformFaultDomainCount.One)
+                    Tags = this.Tags
+                }
+
+            [ hostGroup ]
+
+type HostGroupBuilder() =
+    member _.Yield _ =
+        {
+            Name = ResourceName.Empty
+            AvailabilityZones = None
+            SupportAutomaticPlacement = None
+            PlatformFaultDomainCount = None
+            Tags = Map.empty
+        }
+
+    [<CustomOperation "name">]
+    member _.Name(state: HostGroupConfig, name: string) = { state with Name = ResourceName name }
+
+    [<CustomOperation "add_availability_zones">]
+    member _.AddAvailabilityZone(state: HostGroupConfig, az: string list) =
+        match state.AvailabilityZones with
+        | None ->
+            { state with
+                AvailabilityZones = Some(az |> List.map AvailabilityZone.Parse)
+            }
+        | Some curZones ->
+            { state with
+                AvailabilityZones = Some((az |> List.map AvailabilityZone.Parse) @ curZones)
+            }
+
+    [<CustomOperation "add_availability_zones">]
+    member _.AddAvailabilityZone(state: HostGroupConfig, az: AvailabilityZone list) =
+        match state.AvailabilityZones with
+        | None ->
+            { state with
+                AvailabilityZones = Some az
+            }
+        | Some curZones ->
+            { state with
+                AvailabilityZones = Some(az @ curZones)
+            }
+
+    [<CustomOperation "supportAutomaticPlacement">]
```

Comment:
let's have consistency on customOperation naming, this should be  support_automatic_placement